Zend/zend_vm_def.h

```diff
@@ -4023,6 +4023,10 @@ ZEND_VM_HOT_HANDLER(130, ZEND_DO_UCALL, ANY, ANY, SPEC(RETVAL,OBSERVER))
 	zend_function *fbc = call->func;
 	zval *ret;
 
+	if (UNEXPECTED((fbc->common.fn_flags & ZEND_ACC_DEPRECATED) != 0)) {
```

Comment:
This call should be after `SAVE_OPLINE()`, as user handlers may promote it to exceptions. See https://www.npopov.com/2017/04/14/PHP-7-Virtual-machine.html#exception-handling for details.